src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+interface State {
+  pressedKey: string;
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    pressedKey: '',
+  };
+
+  componentDidMount() {
+    window.addEventListener('keyup', (e: KeyboardEvent) => {
+      this.setState({ pressedKey: e.key });
+    });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keyup', (e: KeyboardEvent) => {
+      this.setState({ pressedKey: e.key });
+    });
+  }
+
+  render() {
+    const { pressedKey } = this.state;
+
+    const infoMessage = pressedKey
+      ? `The last pressed key is [${this.state.pressedKey}]`
```

Comment:
You already destructured this value from this.state
```suggestion
      ? `The last pressed key is [${pressedKey}]`
```